test/e2e/e2e.go

```diff
@@ -558,16 +560,19 @@ func getApiAddress() string {
 }
 
 // IsGatewayModeLocal returns true if the gateway mode is local
-func IsGatewayModeLocal() bool {
-	anno, err := e2ekubectl.RunKubectl("default", "get", "node", "ovn-control-plane", "-o", "template", "--template={{.metadata.annotations}}")
-	if err != nil {
-		framework.Logf("Error getting annotations: %v", err)
-		return false
-	}
-	framework.Logf("Annotations received: %s", anno)
-	isLocal := strings.Contains(anno, "local")
-	framework.Logf("IsGatewayModeLocal returning: %v", isLocal)
-	return isLocal
+func IsGatewayModeLocal(cs kubernetes.Interface) bool {
+	ginkgo.GinkgoHelper()
+	node, err := e2enode.GetRandomReadySchedulableNode(context.TODO(), cs)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	l3ConfigRaw, ok := node.Annotations[util.OvnNodeL3GatewayConfig]
+	if !ok {
+		framework.Failf("Failed to find L3 gateway config for Node %q", node.Name)
+	}
```

Comment:
Same here let's just remove if statements and replace with ginkgo

```suggestion
	l3ConfigRaw, ok := node.Annotations[util.OvnNodeL3GatewayConfig]
	Expect(ok).To(BeTrue(), "should find L3 gateway config for Node %q", node.Name)
```